src/test/java/com/redhat/red/build/finder/DeletedBuildTest.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.redhat.red.build.finder;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.redhat.red.build.koji.KojiClientException;
+import com.redhat.red.build.koji.model.xmlrpc.KojiBuildState;
+
+public class DeletedBuildTest {
+    private BuildFinder finder;
+
+    @Before
+    public void init() {
+        MockKojiClientSession session = new MockKojiClientSession("deleted-build-test");
+        BuildConfig config = new BuildConfig();
+        finder = new BuildFinder(session, config);
+    }
+
+    @Test
+    public void verifyDeletedBuild() throws KojiClientException {
+        Map<String, Collection<String>> checksumTable = new HashMap<>(2);
+
+        checksumTable.put("706c11702729457f7228f3f1ab4d3791", Collections.singletonList("wildfly-core-security-7.5.9.Final-redhat-2.jar"));
+
+        Map<Integer, KojiBuild> builds = finder.findBuilds(checksumTable);
+
+        assertEquals(2, builds.size());
+        assertTrue(builds.containsKey(0));
+        assertTrue(builds.containsKey(505412));
+        assertEquals(KojiBuildState.DELETED, builds.get(505412).getBuildInfo().getBuildState());
+    }
+
+    @Test
+    public void verifyCompletedBuilds() throws KojiClientException {
```

Comment:
This verifies that when we have two completed builds that share the same archive, only the latest (highest id) is kept. Otherwise, we unnecessarily duplicate archives in the previous code in this case.